src/leveled_sstblock.erl

```diff
@@ -0,0 +1,799 @@
+%% -------- SST Block Functions ---------
+%%
+%% Functions to serialise and deserialise blocks in leveled_sst 
+%%
+%% Standard block sizes are
+%% -define(LOOK_BLOCKSIZE, {24, 32}).
+%% -define(NOLOOK_BLOCKSIZE, {56, 32}).
+%% 
+%% Requirement to serialise LOOK_BLOCKS to optimise for picking the nth value
+%% Requirement to serialise NOLOOK_BLOCKS to optimise for picking the first and
+%% last values
+
+-module(leveled_sstblock).
+
+-define(MAX_SUBBLOCK_SIZE, 1 bsl 16).
+-define(BLOCK_TYPE0, 0).
+    % Block is just a list of terms 
+-define(BLOCK_TYPE1, 1).
+    % Lookup block divided into 4 blocks of 6
+    % 24 KV blocks only
+-define(BLOCK_TYPE2, 2).
+    % Lookup block divided into 4 blocks of 8
+    % 32 KV blocks only
+-define(BLOCK_TYPE3, 3).
+    % Nolookup block with first/last terms at head
+-define(BLOCK_TYPE4, 4).
+    % Nolookup block with first/last terms at head, and block split into L/M/R
+    % 56 KV blocks only
+-define(COMPRESSION_FACTOR, 1).
+    % When using native compression - how hard should the compression code
+    % try to reduce the size of the compressed output. 1 Is to imply minimal
+    % effort, 6 is default in OTP:
+    % https://www.erlang.org/doc/man/erlang.html#term_to_binary-2
+-define(BINARY_SETTINGS, [{compressed, ?COMPRESSION_FACTOR}]).
+
+-type block_type() ::
+    ?BLOCK_TYPE0|?BLOCK_TYPE1|?BLOCK_TYPE2|?BLOCK_TYPE3|?BLOCK_TYPE4.
+-type range_filter() ::
+    all|{leveled_codec:ledger_key(), leveled_codec:ledger_key()}.
+-type top_and_tail() ::
+    {
+        leveled_codec:ledger_key()|not_present,
+        leveled_codec:ledger_key()|not_present,
+        fun((range_filter()) -> list(leveled_codec:ledger_kv()))
+    }.
+
+-export(
+    [
+        serialise_block/3,
+        get_all/2,
+        get_topandtail/2,
+        get_nth/3
+    ]
+).
+
+%%%============================================================================
+%%% API
+%%%============================================================================
+
+
+-spec serialise_block(
+    lookup|no_lookup,
+    {0|1, leveled_sst:press_method()},
```

Comment:
```suggestion
    {leveled_sst:block_version(), leveled_sst:press_method()},
```
To make it more consistent and to avoid future changes in just one place

Reply:
Change made